NR-595593: (mobile) Symbol upload script doc update, Bitcode deprecation doc update (#10096)
newrelic/docs-website: src/content/docs/mobile-monitoring/new-relic-mobile-ios/configuration/upload-dsyms-bitcode-apps.mdx

  - Mobile monitoring
  - New Relic Mobile iOS
  - Configuration
metaDescription: How to upload symbols for crash reports from iOS apps monitored with New Relic mobile monitoring.
redirects:
  - /docs/mobile-monitoring/new-relic-mobile-ios/install-configure/upload-dsyms-bitcode-apps
---

Your app's dSYM files are stored in Xcode's dSYM archive path folder. This is the folder where the iOS agent gets the dSYM files that are used to symbolicate your crash reports.
New Relic provides a post-build script as part of the [iOS agent's install process](/docs/mobile-monitoring/new-relic-mobile-ios/installation/spm-installation/#configure-using-swift-package-manager). 
This script automatically converts your dSYM to the New Relic map file format and uploads the files needed for crash symbolication to New Relic.

## Automatic script [#Automatic]

> A Swift script is included with the iOS Agent that is intended to be run from a build script in your target's Build Phases in Xcode.

Reviewer: ```suggestion
A Swift script is included with the iOS Agent that is intended to be run from a build script in your target's build phases in Xcode.
```

In Xcode 14 Apple changed the default setting on newly created projects to disable bitcode. The App Store now only accepts app submissions with bitcode turned off. See [Xcode 14 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes).
If your project still has bitcode enabled you should disable it. The automatic script requires bitcode to be disabled.

Bitcode-enabled apps have their dSYM files generated by Apple. You must [download the dSYM files for Bitcode-enabled apps from Apple](/docs/mobile-monitoring/new-relic-mobile-ios/install-configure/retrieve-upload-dsyms-bitcode-enabled-apps) and upload them to New Relic through the [mobile monitoring UI](#mobile-ui).

If you see unreadable machine code in the [**Crashes** page](/docs/mobile-monitoring/mobile-monitoring-ui/crashes/mobile-apps-crashes-dashboard), your dSYM files may not be uploaded correctly. In some cases, you may need to [manually upload dSYM files](#manual-dsym).
 
In New Relic iOS agent version 7.4.0 we introduced a new symbol upload script which uses Swift. The Python version of the symbol upload script is still available but will be removed in a future release.

> As of New Relic iOS Agent version 7.3.8 the Python version of the automatic script uses Python 3. As of macOS 12.3 [macOS 12.3 (Monterey)](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes) Python 3 will be installed by default with Xcode. If you're using the python automatic script on macOS 12.2 or before, you may need to manually install [Python 3](https://www.python.org/downloads/mac-osx/).

Reviewer: ```suggestion
With the New Relic iOS Agent version 7.3.8, the python version of the automatic script uses Python 3. If you use macOS 12.3 [macOS 12.3 (Monterey)](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes), Python 3 will be installed by default with Xcode. But if you use the python automatic script on macOS 12.2 or before, you may need to manually install [Python 3](https://www.python.org/downloads/mac-osx/).
```

1. Go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Mobile**. Then select your app from the list.
2. View **Crash analysis**.
3. Select a specific crash from the **Crash types** list.
4. Click **Upload dSYM**. You can either drag and drop your dSYMs directly, or select the file from your computer.

## Manually upload dSYM files [#manual-dsym]

> In some circumstances, New Relic's automatic upload of dSYM files may fail. If a dSYM upload is attempted and fails, it creates an `upload_dsym_results` file. For example, if there's a network failure and the dSYM upload isn't completed, `upload_dsym_results` will contain a log of what went wrong. For additional information about how New Relic handles dSYM uploads, see [New Relic's Online Technical Community](https://discuss.newrelic.com/t/relic-solution-ios-dsym-upload-deep-dive/42513).

Reviewer: ```suggestion
In some circumstances, New Relic's automatic upload of dSYM files may fail. If the dSYM upload fails, it creates an `upload_dsym_results` file. For example, if there's a network failure and the dSYM upload isn't completed, `upload_dsym_results` will contain a log of what went wrong. For additional information about how New Relic handles dSYM uploads, see [New Relic's Online Technical Community](https://discuss.newrelic.com/t/relic-solution-ios-dsym-upload-deep-dive/42513).
```

## Troubleshooting [#auto-upload-fail-troubleshooting]

> The automatic script will create an `upload_dsym_results.log` file in your projects root directory containing information about any failures that occur during symbol upload.

Reviewer: ```suggestion
The automatic script will create an `upload_dsym_results.log` file in your project's root directory, which  contains information about any failures that occur during symbol upload.
```
